datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDatePickerViewHolderFactory.kt

```diff
@@ -218,21 +273,39 @@ internal object QuestionnaireItemDatePickerViewHolderFactory :
   }
 }
 
+internal fun handleDateFormatAfterTextChange(
+  editable: Editable,
+  acceptableDateFormat: String,
+  dateFormatSeparator: Char,
+  isDeleting: Boolean
+) {
+  val editableLength = editable.length
+  if (editable.isEmpty()) {
+    return
+  }
+  // restrict date entry upto acceptable date length
+  if (editableLength > acceptableDateFormat.length) {
+    editable.replace(acceptableDateFormat.length, editableLength, "")
+    return
+  }
+  // handle delete text and separator
+  if (editableLength < acceptableDateFormat.length) {
```

Comment:
```suggestion
  if (editable.length < acceptableDateFormat.length) {
```
editable.length will be consistent with acceptableDateFormat.length, can you please update at other places as well.

Reply:
Its used around 10 places , so defined variable for this